pytensor/graph/basic.py

```diff
@@ -907,6 +907,35 @@ def graph_inputs(
     yield from (r for r in ancestors(graphs, blockers) if r.owner is None)
 
 
+def required_graph_inputs(graph):
+    """
+    Get the inputs into PyTensor variables
+
+    Parameters
+    ----------
+        graph: PyTensor `Variable` instances
+        Output `Variable` instances from which to search backward through
+        owners.
+
+    Returns
+    -------
+        List of tensor variables that are input nodes with no owner, in the order
+        found by a left-recursive depth-first search started at the nodes in `graphs`.
+    """
+    from pytensor.compile.sharedvalue import SharedVariable
+
+    if isinstance(graph, tuple | list):
+        graph = graph
```

Comment:
This branch is not doing anything

Reply:
```SharedVariable``` is imported as it is used below in line 935 where we exclude inputs that are SharedVariable and Constants. 
We make graph an iterable because ```graph_inputs``` expects and iterable as its input.
https://github.com/pymc-devs/pytensor/blob/5e612abbc8fe11d9f1a723154521c0a68070c1a9/pytensor/graph/basic.py#L887-L888

Reply:
I'm saying that `graph=graph` is useless, you just need to trigger the else branch, which I guess you can with `if isinstance(graph, Variable)`